### PR TITLE
AAP-30180: trial report, get email/name from RHSSO

### PR DESCRIPTION
**Pull request by goneri**
Jira Issue: <https://issues.redhat.com/browse/AAP-30180>

## Description

Initialize the RHSSO users's email/given_name/family_name from AMS.

### Scenarios tested

Run `wisdom-manage migrate_email_name_from_ams`.

## Production deployment

- :warning:  This code change requires the following considerations before going to production:
    We need to manually run `wisdom-manage migrate_email_name_from_ams` in the secondary environment first.

### CODE REVIEW

**Comment by goneri** [comment]
Converted to Draft until https://github.com/ansible/ansible-ai-connect-service/pull/1281 is merged.

File: ansible_ai_connect/users/migrations/0014_auto_20240829_1738.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+def email_names_from_db(apps, schema_editor):
+    User = apps.get_model("users", "User")
+    for user in User.objects.all():
+        if user.first_name:
+            continue
+        ams = user.get("ams")
+        if not ams:
+            continue
+        user.first_name = ams.get("first_name")
```
Comment (manstis) [inline ansible_ai_connect/users/migrations/0014_auto_20240829_1738.py:14]: You're using `given_name` on your other [PR](https://github.com/ansible/ansible-ai-connect-service/pull/1281/files#diff-d8b9627bf3e9cc38433bebfcb64ea46bc5dee93c7eaa4e4fc62fe6c3fa1d4341R71).

**Review by manstis — COMMENTED** [review]
You probably want to be consistent with `User` property names?

**Comment by jameswnl** [comment]
PR Deployment created in namespace wisdom-pr-1282

 Look up the wisdom-service url in the cluster

File: ansible_ai_connect/users/management/commands/migrate_email_name_from_ams.py
Comment (manstis) [inline ansible_ai_connect/users/management/commands/migrate_email_name_from_ams.py:1]: Out of interest why did you move this from a migration (automatic) to a command (manual)?

I assume we'll need to run this command once this PR is deployed. 

Is it  "one off" or do you foresee a need to schedule it?

**Review by manstis — APPROVED** [review]
I'm intrigued by the command; but LGTM :+1:

File: ansible_ai_connect/users/management/commands/migrate_email_name_from_ams.py
Comment (goneri) [reply ansible_ai_connect/users/management/commands/migrate_email_name_from_ams.py:1]: I discovered that the DB migration scripts have only access to the DB model object properties, not the whole application. To properly do the migration I need to access the authz_check instance which rely on the application itself.
